sdks/java/extensions/timeseries/src/main/java/org/apache/beam/sdk/extensions/timeseries/joins/BiStreamJoinResultCoder.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.timeseries.joins;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.beam.sdk.coders.*;
+import org.apache.beam.sdk.values.TimestampedValue;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.sdk.values.TypeParameter;
+
+public class BiStreamJoinResultCoder<K, V1, V2>
+    extends StructuredCoder<BiTemporalJoinResult<K, V1, V2>> {
+
+  private final Coder<V1> leftCoder;
+  private final Coder<V2> rightCoder;
+  private final Coder<K> keyCoder;
+
+  public BiStreamJoinResultCoder(Coder<K> keyCoder, Coder<V1> leftCoder, Coder<V2> rightCoder) {
+    this.leftCoder = NullableCoder.of(leftCoder);
+    this.rightCoder = NullableCoder.of(rightCoder);
+    this.keyCoder = NullableCoder.of(keyCoder);
+  }
+
+  public Coder<V1> getLeftCoder() {
+    return leftCoder;
+  }
+
+  public Coder<V2> getRightCoder() {
+    return rightCoder;
+  }
+
+  public Coder<K> getKeyCoder() {
+    return keyCoder;
+  }
+
+  public static <K, V1, V2> BiStreamJoinResultCoder<K, V1, V2> of(
+      Coder<K> keyCoder, Coder<V1> leftCoder, Coder<V2> rightCoder) {
+
+    return new BiStreamJoinResultCoder<>(keyCoder, leftCoder, rightCoder);
+  }
+
+  @Override
+  public void encode(BiTemporalJoinResult<K, V1, V2> value, OutputStream outStream)
+      throws IOException {
+
+    if (value == null) {
```

Comment:
key, left and right coders are already `NullableCoder` which tolerates `NULL`?

Reply:
BiStreamJoinResultCoder file was left over artifact from rename of class. Thanx for catching.